Userland/DevTools/HackStudio/HackStudioWidget.cpp

```diff
@@ -302,26 +304,28 @@ Vector<DeprecatedString> HackStudioWidget::selected_file_paths() const
     return files;
 }
 
-bool HackStudioWidget::open_file(DeprecatedString const& full_filename, size_t line, size_t column)
+ErrorOr<void> HackStudioWidget::open_file(DeprecatedString const& full_filename, size_t line, size_t column)
 {
     DeprecatedString filename = full_filename;
     if (full_filename.starts_with(project().root_path())) {
         filename = LexicalPath::relative_path(full_filename, project().root_path());
     }
-    if (Core::File::is_directory(filename) || !Core::File::exists(filename))
-        return false;
+    if (Core::File::is_directory(filename))
+        return Error::from_errno(EISDIR);
+    if (!Core::File::exists(filename))
+        return Error::from_errno(ENOENT);
```

Comment:
I'm not 100% sure on this, but I think `from_errno()` is supposed to only be used by POSIX functions that would otherwise set `errno`. Here, it would be better to pass a string error message.

Reply:
Seems like a reasonable assumption to make, my thinking behind this was that we already had the messages baked in somewhere rather than just returning a string I made up.